typed-racket-lib/typed-racket/base-env/base-types-extra.rkt

```diff
@@ -17,7 +17,7 @@
 (define-other-types
   -> ->* case-> U Rec All Opaque Vector
   Parameterof List List* Class Object Values Instance Refinement
-  pred Struct Struct-Type Prefab Top Bot)
+  pred Struct Struct-Type Prefab Top Bot Distinction)
```

Comment:
I couldn't think of any other way to allow both new-subtypes to refer to type aliases and for other forms to refer to new-subtypes properly. 

But actually, this still is broken, in this example:
```racket
#lang typed/racket/base
(define-new-subtype Lst (make-Lst (U Null (Pairof Elt Lst))))
(define-new-subtype Elt (make-Elt (U Symbol Lst)))
(: cns : Elt Lst -> (Pairof Elt Lst))
(define cns
  (inst cons Elt Lst))
(: lst : Elt * -> Lst)
(define (lst . args)
  (for/fold ([lst (make-Lst '())]) ([elt (in-list (reverse args))])
    (make-Lst (cns elt lst))))
```
With this error highlighting the `lst` in the `for/fold` accumulator:
```
. Type Checker: type mismatch
  expected: Lst
  given: Lst in: lst
```